### DISCUSSION TITLE
Problem booting & EEPROM wiped ?

### DISCUSSION THREAD

**Discussion by dynek** [gcormier/megadesk#120] [General]
Hello,

I've been happily using a `megadesk` for quite some months now and from time to time, I can hear the initialization sound running over and over again, is there something I can do to avoid a boot loop ? Should I leave my desk power-on all the time ? I am actually turning off when I'm done working.

Also very recently, I wanted to recall a given height and all of them seem to be wiped, here's the sound I can hear:
https://user-images.githubusercontent.com/591091/210353427-b72c1bf9-1bc9-47c8-8260-bd2a6312a285.mp4

Thank you

**Comment by gcormier** [comment]
If it is looping, it's failing to handshake with the motors. Unless you really want to turn off the desk, leaving it on will be fine as it will consume negligible power. The motors do not require any power to hold the position (obviously, as the desk does not drop when unplugged). I do know the handshake will fail from time to time, but normally within 2 loops of the beeps, it will sync.

You could try doing a re-calibration of the motors. I think any difference in the motor positions can sometimes cause issues, but I have never looked into it further.

As to it resetting eeprom, that shouldn't happen. The sound you attached is the sound when trying to recall a memory position that has not been saved. The only thing that will trigger an EEPROM reset is holding one of the buttons while powering on.

The EEPROM symptom is very odd, and I can't help but wonder if the microcontroller has been damaged somehow. I see you ordered 2 units, is the 2nd unit having any issues? Could you try swapping them?

**Reply by dynek** [reply]
Thank you for your reply.

I will leave the desk on all the time. I was using a multi-socket mains extension to turn on/off multiple things.

Regarding the EEPROM, I'm pretty sure I never held a button while powering the desk on, the extension and buttons are at the opposite of each other. Not a big issue, I reconfigured the different heights and I'll see how it behaves.
The second `megadesk` went to a colleague of mine who hasn't installed it yet (@xens 😉).

**Reply by tyler-santos** [reply]
I am seeing this same problem - I have many items (my 2 displays, macbook power, thunderbolt hub, my desk and so on), all plugged into a surge bar. This is attached to a smart plug so that I am not vampire sucking power over night on all these devices (automation turns everything off at night), the last time I went to use my desk the memory positions were forgotten and the only thing I have changed is that it powers off over night now. Worst case scenario I will just separate the desk plug from the rest but it would be nice to keep them together. I will see if it happens again, so far it has only happened once but I've only had the smart plug setup for a few days.

**Comment by dynek** [comment]
So far so good, everything works properly. Thank you @gcormier

**Reply by gcormier** [reply]
Excellent! Please let me know if the issue reappears.

**Comment by steilerDev** [comment]
@gcormier I also recently got one of your controllers - I got a 'smart outlet' at my desk which turns of power to the whole desk (including my PC/dock etc.)

I get the same sound as dynek after the desk lost power - is this expected or is there a way to persist the configuration between power losses?

**Reply by gcormier** [reply]
The settings are in EEPROM, so they will survive without power. So this is unexpected behavior.

Does it lose your settings every time the power is off?

**Reply by steilerDev** [reply]
I'd say yes (i.e. everytime I try to recall the setting it's gone :D ) - I will monitor the situation to see if the behaviour is consistent

**Reply by gcormier** [reply]
Really odd. I can't reproduce this issue on mine, I will grab another board this weekend to play with.

Can you record your startup sound?

**Reply by steilerDev** [reply]
hmm - so I've been monitoring and it seems since complaining about it, he did no longer loose the configuration :D

**Reply by gcormier** [reply]
Well, good to hear, but intermittent issues are the worst kind!

**Reply by steilerDev** [reply]
Yes they are - anything I should look out for?

**Reply by steilerDev** [reply]
Lost configuration the last two nights again :(

**Reply by gcormier** [reply]
Darn.

Did you remove the power? Or it was powered on the whole time and it just forgot?

Does it lose the memory positions, or everything else? (eg. if you had set it to silent mode, does it revert?)

**Reply by steilerDev** [reply]
It's loosing power every night as it's connected to the same outlet as the rest of my desk setup (and I unfortunately don't have a separate outlet available that is powered all the time)

It's at least loosing memory positions - haven't really configured anything else.

**Reply by gcormier** [reply]
Can you record the startup sounds when powering it on?

And, would you mind setting a different mode and see if it loses that as well?

The startup sounds will differ slightly as follows

1. When the EEPROM has not yet been initialized (the first time it is powered on after being programmed), there will be 3 beeps, and then it will clear the entire EEPROM which takes about 1 second. Then the 4th beep will occur.
2. When the EEPROM has been initialized, it will not do anything EEPROM clear - so there will be 3 beeps, and then a brief delay while it handshakes with the motors, and then the 4th.  So that delay before the 4th beep is much shorter, maybe 250ms.

**Reply by steilerDev** [reply]
[Yesterday](https://user-images.githubusercontent.com/7031616/212621305-78c59880-79ab-44b1-890a-70c6e6e0eb78.mp4) and [today](https://user-images.githubusercontent.com/7031616/212621308-ef32640e-156a-4a71-aa81-eca1fa9b17db.mp4) I recorded the startup sound (please ignore the startup sounds of my various other things attached to the outlet). 

In  both cases the unit lost configuration, and looking at your explanation, I think it is initializing the EEPROM - any idea what could be wrong/how to fix this?

**Comment by Yveaux** [comment]
#169 may fix the eeprom loosing settings for you.
